agent/agent_endpoint.go

```diff
@@ -501,14 +510,27 @@ func (s *HTTPServer) AgentHealthServiceId(resp http.ResponseWriter, req *http.Re
 	for _, service := range services {
 		if service.ID == serviceID {
 			code, status := AgentHealthService(serviceID, s, resp, req)
+			if returnTextPlain(req) {
+				resp.WriteHeader(code)
+				fmt.Fprint(resp, status)
+				return nil, nil
+			}
+			resp.Header().Add("Content-Type", "application/json")
```

Comment:
No need for this as its done in http.go

Reply:
No, this is actually needed: since we return non HTTP 200 Error codes, If we don't do that, the Mime Type won't be properly sent (because HTTP Status Code has already been sent)

I did not find a way to both specify the mime-type, having errors properly specified (ie: having correct JSON format) and having the proper error code.

What do you suggest?